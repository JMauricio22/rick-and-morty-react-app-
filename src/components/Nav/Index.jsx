import React from "react";
import { Navbar, Container, Nav } from "react-bootstrap";
import { Link } from "react-router-dom";

export default function Index() {
  return (
    <Navbar bg='dark' variant='dark' expand='lg'>
      <Container>
        <Navbar.Brand to='/' as={Link}>
          Rick And Morty
        </Navbar.Brand>
        <Navbar.Toggle aria-controls='basic-navbar-nav' />
        <Navbar.Collapse id='basic-navbar-nav'>
          <Nav className='ms-auto'>
            <Nav.Link to='/' as={Link}>
              Home
            </Nav.Link>
            <Nav.Link to='/favorites' as={Link}>
              Favorites
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}
