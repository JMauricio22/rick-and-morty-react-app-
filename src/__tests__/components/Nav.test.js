import { render, screen } from "@testing-library/react";
import Nav from "../../components/Nav/Index";
import { BrowserRouter } from "react-router-dom";

describe("<Nav />", () => {
  test("Render Nav", () => {
    render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>
    );
    expect(screen.getByRole("navigation")).toBeInTheDocument();
    expect(screen.getByRole("navigation")).toMatchSnapshot();
  });
});
