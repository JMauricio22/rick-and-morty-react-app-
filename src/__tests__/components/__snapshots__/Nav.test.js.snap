// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> Render Nav 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    class="container"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Rick And Morty
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="ms-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/favorites"
        >
          Favorites
        </a>
      </div>
    </div>
  </div>
</nav>
`;
